pkg/openSearch/openSearchClient/client.go

```diff
@@ -45,27 +46,32 @@ func NewClient(openSearchProjectClient *opensearch.Client, updateMaxRetries int,
 // requestBody is the request body to send to OpenSearch.
 // It returns the response body as or an error in case something went wrong.
 func (c *Client) Search(indexName string, requestBody []byte) (responseBody []byte, err error) {
-	log.Debug().Msgf("search requestBody: %s", string(requestBody))
-	searchResponse, err := c.openSearchProjectClient.Search(
-		c.openSearchProjectClient.Search.WithIndex(indexName),
-		c.openSearchProjectClient.Search.WithBody(bytes.NewReader(requestBody)),
+	log.Debug().Str("src", "opensearch").Msgf("request: %s", string(requestBody))
+
+	req, err := c.openSearchProjectClient.Search(
+		context.Background(),
```

Comment:
I think we can pass the context as an argument in the function signature, allowing the caller to decide whether to run it in the background or handle cancellation.
`Search(ctx context.Context, indexName string, requestBody []byte) (responseBody []byte, err error)`


Similar observation where we interact with opensearch APIs

Reply:
@endyApina there has been discussion to use `context.Background()` instead of passing a context from the caller  in the previous comments. 

https://github.com/greenbone/opensight-golang-libraries/pull/84#pullrequestreview-2145936451

Changing the signature would also break the wrapper and the interface would have to be changed.